### PR TITLE
docs: improve developer debugging instructions

### LINKED ISSUE #1207: Document how to use a debugger with aks-engine
Using VS Code to debug `aks-engine` interactively is a bit tricky to set up but works great. Let's document it.

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

Updates the developer documentation about debugging with VS Code and Delve so it works and has pretty pictures.

**Issue Fixed**:

Fixes #1207 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: docs/community/developer-guide.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 
-- [Go](https://golang.org/dl) 1.11 or later
-- Golang [dep](https://github.com/golang/dep) 0.5.0 or later
-- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) 1.7 or later
-- An Azure account (needed for deploying VMs and Azure infrastructure)
-- Git
+- [`git`](https://git-scm.com/)
+- [`go`](https://golang.org/dl) 1.12 or later
+- [`dep`](https://github.com/golang/dep) 0.5.1 or later
+- An [Azure](https://azure.microsoft.com/en-us/) account
```
Comment (craiglpeters) [inline docs/community/developer-guide.md]: Usually, it is called an Azure Subscription

File: docs/community/developer-guide.md [hunk 2]
```diff
@@ -206,47 +205,104 @@ matches the version of the Kubernetes server.
...
+        "--set", "servicePrincipalProfile.secret=my-service-principal-secret"
+      ],
       "showLog": true
     }
   ]
 }
 ```
 
+Copy and paste the configuration and change its name and details as needed. It's easy to build up
+a `launch.json` file to debug `aks-engine upgrade`, `scale`, and other commands and variations.
```
Comment (craiglpeters) [inline docs/community/developer-guide.md]: Easy? For whom? What do you need to understand to make it easy?

**Comment by jackfrancis** [comment]
This is awesome, thank you @mboersma!

File: docs/community/developer-guide.md (hunk 2 above)
Comment (mboersma) [reply docs/community/developer-guide.md]: I meant this to be encouraging, but you're right that it's not "easy" in any objective sense. I'll rewrite this.

**Comment by jackfrancis** [comment]
lgtm

**Pull request merged**